dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java

```diff
@@ -231,7 +245,7 @@ private JerseyRequestHttpEntity(ClientRequest clientRequest) {
          */
         @Override
         public boolean isRepeatable() {
-            return false;
+            return isEntityRepeatable.test(clientRequest);
```

Comment:
I think this will not work because Jersey's `ClientRequest` can't write the entity twice. I think only raw byte arrays should be repeatable if there is no gzip used.

Reply:
```Java
    private static class JerseyRequestHttpEntity extends AbstractHttpEntity {

        private ClientRequest clientRequest;
        private final boolean gzipEnabled;

        private JerseyRequestHttpEntity(ClientRequest clientRequest, boolean gzipEnabled) {
            this.clientRequest = clientRequest;
            this.gzipEnabled = gzipEnabled;
            setChunked(true);
        }

        /**
         * {@inheritDoc}
         */
        @Override
        public boolean isRepeatable() {
            return !gzipEnabled && clientRequest.getEntityClass() == byte[].class;
        }

        /**
         * {@inheritDoc}
         */
        @Override
        public long getContentLength() {
            return -1;
        }

        /**
         * {@inheritDoc}
         * <p>
         * This method isn't supported at will throw an {@link java.lang.UnsupportedOperationException}
         * if invoked.
         * </p>
         */
        @Override
        public InputStream getContent() throws IOException {
            // Shouldn't be called
            throw new UnsupportedOperationException("Reading from the entity is not supported");
        }

        /**
         * {@inheritDoc}
         */
        @Override
        public void writeTo(final OutputStream outputStream) throws IOException {
            if (isRepeatable() && clientRequest.isCommitted()){
                byte[] data = (byte[]) clientRequest.getEntity();
                outputStream.write(data);
            } else {
                clientRequest.setStreamProvider(contentLength -> outputStream);
                clientRequest.writeEntity();
            }
        }
```
That way we can pass the predicate.

Reply:
That's a mess. Let me think it over.